src/onINP.ts

```diff
@@ -57,20 +63,88 @@ const getInteractionCountForNavigation = () => {
 const MAX_INTERACTIONS_TO_CONSIDER = 10;
 
 // A list of longest interactions on the page (by latency) sorted so the
-// longest one is first. The list is as most MAX_INTERACTIONS_TO_CONSIDER long.
+// longest one is first. The list is at most MAX_INTERACTIONS_TO_CONSIDER long.
 let longestInteractionList: Interaction[] = [];
 
 // A mapping of longest interactions by their interaction ID.
 // This is used for faster lookup.
-const longestInteractionMap: {[interactionId: string]: Interaction} = {};
+const longestInteractionMap: {[interactionId: string]: Interaction} =
+  Object.create(null);
+
+// A mapping between a particular frame's render time and all of the
+// event timing entries that occurred within that frame. Since event timing
+// entries round their duration to the nearest 8sm, this ends up being
+// a best-effort guess, but but debugging and attribution practices it's
+// useful to see all events the occurred within the same frame as a particular
+// interaction, so it's worth doing the best-effort guess.
+// const pendingEntriesGroupMap: Map<number, PendingEntriesGroup> = new Map();
+const pendingEntriesGroupMap: {[renderTime: string]: PendingEntriesGroup} =
+  Object.create(null);
```

Comment:
FYI: I'm using `Object.create(null)` here (rather than `{}`) because I need to use a `for...in` loop later and I don't want to have to add an `Object.hasOwnProperty()` check. And the reason I need a `for...in` loop is I want to be able to break out early. And the reason I don't want to use a `for...of` loop is because when that gets transpiled to ES5 it adds a bunch of Babel helpers, so this is the easiest solution...

Reply:
Should we add some comment to the code to explain this?

And do we need line 80 which is commented out?

Reply:
Ahh, I just got tripped up by the `Object.create(null)` further up, as I'd never seen it before, and it caught my attention, so a comment might make sense.

Reply:
> FYI: I'm using Object.create(null) here (rather than {}) because I need to use a for...in loop later and I don't want to have to add an Object.hasOwnProperty() check. And the reason I need a for...in loop is I want to be able to break out early. And the reason I don't want to use a for...of loop is because when that gets transpiled to ES5 it adds a bunch of Babel helpers, so this is the easiest solution...

Does a `Map` scratch all the relevant itches?

You'd need an `Array.from` call to use it with `for in` loop, but here's what it could look like:

```es6
const keys = Array.from(map.keys());

for (key in keys) {
  console.log(key)
  break;
}
```

Reply:
@vanderhoop I think the biggest issue with any alternate approach is IE9 support without pulling in a bunch of Babel code.

@philipwalton `Object.keys()` and then a classic `for (let i = 0; i < keys.length; i++) { const key = keys[i]; ...}` loop could also work, and would avoid surprising readers of the code and the need for a comment. It's also somewhat faster than `for...in` (in microbenchmarks), though either way seems fine.

Reply:
I updated the implementation and `Object.create(null)` is no longer used, so I'm going to resolve this.